File: Python/Matrix/4_row_with_max_number_of_1s.py
"""
Idea:
counting no. of one's in each row, and storing the value of row, which has the highest count.
"""


class Solution:

    def rowWithMax1s(self, arr, n, m):
        ans = -1
        max = 0
        for i in range(n):
            count = 0
            for j in range(m):
                if arr[i][j] == 1:
                    count += 1

            if max < count:
                max = count
                ans = i

        return ans


if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n, m = list(map(int, input().strip().split()))

        inputLine = list(map(int, input().strip().split()))
        arr = [[0 for j in range(m)] for i in range(n)]

        for i in range(n):
            for j in range(m):
                arr[i][j] = inputLine[i * m + j]
        ob = Solution()
        ans = ob.rowWithMax1s(arr, n, m)
        print(ans)
        tc -= 1

File: C++/Matrix/4_row_with_max_number_of_1s.cpp
/*
Idea:
counting no. of one's in each row, and storing the value of row, which has highest count.
*/
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int rowWithMax1s(vector<vector<int> > arr, int n, int m) {
        // code here
        int max = 0, index = -1;
        for (int i = 0;i < n;++i)
        {
            int count = 0;
            for (int j = 0;j < m;++j)
                if (arr[i][j] == 1)
                    ++count;

            if (max < count)
            {
                max = count;
                index = i;
            }
        }
        return index;
    }

};

int main()
{
    int t;
    cin >> t;
    while (t--) {
        int n, m;
        cin >> n >> m;
        vector< vector<int> > arr(n, vector<int>(m));
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                cin >> arr[i][j];
            }
        }
        Solution ob;
        auto ans = ob.rowWithMax1s(arr, n, m);
        cout << ans << "\n";
    }
    return 0;
}
